btray/forms.py

```diff
@@ -34,6 +34,23 @@ def validate(self):
         self.user = user
         return True
 
+class SignUpForm(Form):
+    username = TextField('Username', [validators.Required()])
+    email = TextField('Email', [validators.Required()])
+    password = PasswordField('Password', [validators.Required()])
+    submit = SubmitField('Sign Up')
+
+    def validate(self):
+        if not Form.validate(self):
+            return False
+
+        user = User.query.filter_by(username=self.username.data).first()
```

Comment:
I would see if you can replace with `.any()` here.

Reply:
Good call.